confidential-data-hub/hub/src/hub.rs

```diff
@@ -93,19 +55,13 @@ impl DataHub for Hub {
 
     async fn get_resource(&self, uri: String) -> Result<Vec<u8>> {
         // to initialize a get_resource_provider client we do not need the ProviderSettings.
-        let mut client = kms::new_getter(&self.get_resource_provider, ProviderSettings::default())
+        let mut client = kms::new_getter("kbs", ProviderSettings::default())
```

Comment:
Any `sev` option here?

Also, why create a new getter for each request? That means that you have to use the fake client. Instead could we just create the getter once at the beginning?

Reply:
the concrete kbc under `kbs` plugin is determined by `aa_kbc_params`. So we do not need to care if it uses `online-sev-kbc` of `cc-kbc`.

> Also, why create a new getter for each request? That means that you have to use the fake client. Instead could we just create the getter once at the beginning?

Your idea is ok, too. But we will store an extra fake client object in CDH, while the fake client only provides the function pointer to the real client. Also, when an unseal secret request with `kbs` provider comes, it will still create a fake client. The consumption of creation a fake client is very low (check the lock), thus I think the way makes sense.

Reply:
Ok, the first part is a bit confusing because in the past the KbcClient was just for the KBS protocol, but here it can do both that and the SEV protocol. I guess that is ok tho.